edx_exams/apps/core/api.py

```diff
@@ -48,6 +48,27 @@ def get_latest_attempt_for_user(user_id):
     return latest_attempt
 
 
+def get_user_by_anonymous_id(anonymous_user_id, attempt_number, resource_id):
```

Comment:
This function name seems to imply this returns the user for an anonymous id but it in fact returns an exam attempt. Overall comment on this is similar to my previous tho. Any way we can shoot for some code reuse here? We already have an api function that gives us an attempt for user, resource_id, and attempt_number which has near identical code.

If this just returned the user we could reuse that existing function based on the id of that user.